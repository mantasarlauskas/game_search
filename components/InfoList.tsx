import React, { Fragment } from 'react';
import { Info } from 'utils/types';

function InfoList({ list }: InfoListProps) {
    return (
        <>
            {list.map(({ name, slug }, idx) => (
                <Fragment key={slug}>
                    {idx < list.length - 1 ? `${name}, ` : name}
                </Fragment>
            ))}
        </>
    );
}

interface InfoListProps {
    list: Info[];
}

export default InfoList;
