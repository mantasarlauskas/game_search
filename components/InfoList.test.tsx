import { render, screen } from '@testing-library/react';
import InfoList from 'components/InfoList';
import userEvent from '@testing-library/user-event';

describe('<InfoList />', () => {
    const list = [
        { name: 'name1', id: 1 },
        { name: 'name2', id: 2, url: 'url2' },
    ];

    it('matches snapshot', () => {
        const { container } = render(<InfoList list={list} />);
        expect(container).toMatchSnapshot();
    });

    it('shows info list and on click opens correct page', () => {
        render(<InfoList list={list} />);
        userEvent.click(screen.getByText('name2'));
        expect(window.location.href).toEqual('url2');
        userEvent.click(screen.getByText(/name1/));
        expect(window.location.href).toEqual('url2');
    });
});
