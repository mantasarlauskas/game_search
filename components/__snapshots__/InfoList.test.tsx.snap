// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InfoList /> matches snapshot 1`] = `
<div>
  name1, 
  <div>
    <span
      class="item"
    >
      name2
    </span>
  </div>
</div>
`;
